src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -224,37 +268,43 @@ void GcsAutoscalerStateManager::GetNodeStates(
         gcs_node_info.state_snapshot().node_activity());
 
     // The node is alive. We need to check if the node is idle.
-    auto const &node_resource_data = cluster_resource_manager_.GetNodeResources(
-        scheduling::NodeID(node_state_proto->node_id()));
-    if (node_resource_data.is_draining) {
+    auto const node_resource_iter =
+        node_resource_info_.find(NodeID::FromBinary(node_state_proto->node_id()));
+
+    // The node has been added to GcsInfo but we are missing reporting for it. Ignore.
+    if (node_resource_iter == node_resource_info_.end()) {
+      return;
```

Comment:
I'm a bit concerned about this: if we early return, the proto will contain all the default values. Is that correct? How does the current node handle this case?

Reply:
Looks like the unhandled edge case was pre-existing in the code, but when it was pulling from `cluster_resource_manager`, it was less likely to happen since it would just silently return stale data as long as at least one Ray Syncer message from the node was successfully received.

Now if we don't get any responses to the polling (interval 10x longer than syncer), we could run into this.

To handle this case I set the node state to `UNSPECIFIED`, I feel like the autoscaler-side work item can be separate? cc @rickyyx to confirm pre-existing behavior

Reply:
Before we will always have the node (otherwise map look fails and raises): https://github.com/ray-project/ray/blob/a77751707b528ecd3e55f580c415833d4179f4f1/src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc#L227-L228

Reply:
I think this should be a check?

```
RAY_CHECK(node_resource_iter != node_resource_info_.end())
```

Reply:
It could be transient network error though